### PR TITLE
feat(types): Make DepositPreauth easier to guard

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

`DepositPreauth` transaction is now a union type to correctly model that the `Authorize` and `Unauthorize` fields cannot appear at the same time.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation Updates

### CODE REVIEW

File: packages/xrpl/HISTORY.md [hunk 1]
```diff
@@ -13,6 +13,9 @@ Subscribe to [the **xrpl-announce** mailing list](https://groups.google.com/g/xr
 * Add type for NFToken object that is stored on a `NFTokenPage`.
 * Add type for `account_info`'s `account_flags` property.
 
+### Changed
+* `DeposityPreauth` has a new type definition that will not allow `Authorize` and `Unauthorize` on the same instance. This provides better guard support.
```
Comment (mvadari) [inline packages/xrpl/HISTORY.md:17]:
```suggestion
* `DepositPreauth` has a new type definition that will not allow `Authorize` and `Unauthorize` on the same instance. This provides better guard support.
```

File: packages/xrpl/src/models/transactions/index.ts [hunk 2]
```diff
@@ -10,7 +10,11 @@ export { AccountDelete } from './accountDelete'
 export { CheckCancel } from './checkCancel'
 export { CheckCash } from './checkCash'
 export { CheckCreate } from './checkCreate'
-export { DepositPreauth } from './depositPreauth'
+export {
+  DepositPreauth,
+  DepositPreauthWithAuthorize,
+  DepositPreauthWithUnauthorize,
```
Comment (mvadari) [inline packages/xrpl/src/models/transactions/index.ts:16]: Do these two need to be exported?

Comment (ckniffen) [reply packages/xrpl/src/models/transactions/index.ts:17]: Yes if people want to type guard.

**Comment by ckniffen** [comment]
This makes the documentation output look goofy.  Closing for now.

**Pull request closed without merging**